### PR TITLE
Final phase of WOR-1250: pass resources by name to WSM

### PR DESCRIPTION
**Pull request by rtitle**
https://broadworkbench.atlassian.net/browse/WOR-1250

## Summary of changes

Final phase of WOR-1250. Now that the client is updated and https://github.com/DataBiosphere/terra-workspace-manager/pull/1467 is merged, simplify some Leo code by passing resources by-name.

### What

-

### Why

- This was a multi-phase effort to simplify some WSM/Leo interaction, and make WSM resource deletion more robust and less interdependent.

### What to test

- 

### Who tested and where

- [ ] This change is covered by automated tests 
  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
- [ ] I validated this change 
- [ ] Primary reviewer validated this change 
- [ ] I validated this change in the dev environment

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala [hunk 1]
```diff
@@ -815,20 +814,6 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
         case _                         => namespacePrefix
       }
 
-      // Obtain external database IDs by querying WSM and filtering by name
```
Comment (LizBaldo) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala:818]: 🎉

**Review by LizBaldo — APPROVED** [review]
This is so nice, thank you!

**Comment by rtitle** [comment]
I successfully tested this on a BEE, will merge when CI is green

**Comment by rtitle** [comment]
```
org.broadinstitute.dsde.workbench.leonardo.apps.AppLifecycleSpec.create CROMWELL app, delete it and re-create it with same disk

ERROR was not equal to RUNNING
```

This is a GCP test, so feels unrelated… jenkins retest anyway

**Pull request merged**